Fetcher docs should make it clear what needs to be implemented

Should add a sentence in FetcherInterface docstring that explains that the public API of the interface is implemented already: implementations of the interface only need to implement the internal

    def _fetch(self, url: str) -> Iterator[bytes]

I keep encountering this `make: *** No rule to make target 'html'.  Stop.` error when trying to run `make html` from the terminal. According to a few solutions online, I found that a certain 'Makefile' would be needed in the folder where I'm running `make html` from.

--- a/docs/api/tuf.ngclient.fetcher.rst
+++ b/docs/api/tuf.ngclient.fetcher.rst
@@ -3,3 +3,4 @@ Fetcher
 
 .. automodule:: tuf.ngclient.fetcher
    :undoc-members:
+   :private-members: _fetch

--- a/tuf/ngclient/fetcher.py
+++ b/tuf/ngclient/fetcher.py
@@ -23,6 +23,9 @@ class FetcherInterface:
     By providing a concrete implementation of the abstract interface,
     users of the framework can plug-in their preferred/customized
     network stack.
+
+    Implementations of FetcherInterface only need to implement ``_fetch()``.
+    The public API of the class is already implemented.
     """
 
     __metaclass__ = abc.ABCMeta

